### PR TITLE
Give top privilege to logger in debug builds

### LINKED ISSUE #1751: Set log node declassification privilege to "top" in `oak_debug` builds
This will allow applications running in debug mode to create the logger as with top secret confidentiality, which would allow all nodes to successfully log during debug runs.

The SDK should also be changed to try to create the log node as top secret (which would fail in non-debug builds) and fall back to creating a public log node if that does not work.

### PR DESCRIPTION
**Pull request by conradgrobler**
Fixes #1751 

# Checklist

- [x] Pull request affects core Oak functionality (e.g. runtime, SDK, ABI)
  - [ ] I have written tests that cover the code changes.
  - [ ] I have checked that these tests are run by
        [Cloudbuild](/cloudbuild.yaml)
  - [ ] I have updated [documentation](/docs/) accordingly.
  - [ ] I have raised an [issue](https://github.com/project-oak/oak/issues) to
        cover any TODOs and/or unfinished work.
- [ ] Pull request includes prototype/experimental work that is under
      construction.

### CODE REVIEW

File: oak_runtime/src/node/logger.rs [hunk 1]
```diff
@@ -46,6 +46,12 @@ impl super::Node for LogNode {
         "logger"
     }
 
+    #[cfg(feature = "oak_debug")]
+    fn get_privilege(&self) -> NodePrivilege {
+        // Allow the logger Node to declassify log messages in debug builds only.
+        NodePrivilege::top_privilege()
+    }
```
Comment (tiziano88) [inline oak_runtime/src/node/logger.rs:53]: Would it make sense to explicitly have get_privilege returning the empty privilege here for non-debug? Or do you think it's sufficiently clear that it is the default? Maybe just mention the default?

File: sdk/rust/oak/src/logger/mod.rs [hunk 2]
```diff
@@ -82,9 +85,20 @@ pub fn init(log_sender: Sender<LogMessage>, level: Level) -> Result<(), SetLogge
 }
 
 pub fn create() -> Result<Sender<LogMessage>, OakStatus> {
-    crate::io::node_create(
+    match crate::io::node_create(
```
Comment (tiziano88) [inline sdk/rust/oak/src/logger/mod.rs:88]: (optional) Alternative more functional way:

```rust
for label in [top, public] {[
  match node_create(...) {
    Err(PermissionDenied) => continue,
    Err(err) => return Err(err),
    Ok(n) => return Ok(n),
  }
}
Err(...)
```

**Review by tiziano88 — APPROVED** [review]

File: oak_runtime/src/node/logger.rs (hunk 1 above)
Comment (conradgrobler) [reply oak_runtime/src/node/logger.rs:53]: Added an explict implementation to return the empty privilege for non-debug builds.

File: sdk/rust/oak/src/logger/mod.rs (hunk 2 above)
Comment (conradgrobler) [reply sdk/rust/oak/src/logger/mod.rs]: Done.

**Pull request merged**